core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -241,9 +241,9 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         }
     }
 
-    private void failWithUnexpectedMessageType(ChannelHandlerContext ctx, Object msg) {
+    private void failWithUnexpectedMessageType(ChannelHandlerContext ctx, Object msg, Class<?> expected) {
         fail(ctx, new ProtocolViolationException(
-                "unexpected message type: " + msg.getClass().getName()));
+                "unexpected message type: " + msg.getClass().getName() + ", expected " + expected.getName()));
```

Comment:
How about: `... (expected: ...)` for consistency with the `IllegalArgumentException` messages?

Reply:
sgtm, fixed